atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheIntegrationTest.java

```diff
@@ -287,11 +304,13 @@ public void nonContiguousEventsThrows() {
 
     @Test
     public void upToDateClientDoesNotThrow() {
+        eventCache = createEventCache(6);
         setupInitialState();
         eventCache.processStartTransactionsUpdate(TIMESTAMPS_2, SUCCESS);
-        assertThat(eventCache.getUpdateForTransactions(
-                TIMESTAMPS,
-                Optional.of(LockWatchVersion.of(LEADER, SUCCESS_VERSION))).events()).isEmpty();
+        assertThatCode(() -> eventCache.getUpdateForTransactions(
+                ImmutableSet.of(START_TS, 16L),
+                Optional.of(LockWatchVersion.of(LEADER, 3L))))
```

Comment:
Hmm, I feel like I'd rather keep it as an assertion, and maybe rename the test? And if you add the "latestVersion" to TransactionUpdate that can tighten this test as well?

Reply:
I've actually changed the test:
- before, it was testing that a client that is on the exact same version as the cache will not throw
- now, it's testing that a client that is on the same version as the transaction (but _not_ the cache) will not throw.

It'd probably be best here if I make sure we have both of those tests.